### PR TITLE
Sample Updates: SAS Telemetry and Twin; Logging

### PR DESCRIPTION
**Pull request by momuno**
- Changes name of Logging Macros to match C Guidelines. Updates in all sample files.
- Updates Sas Telemetry sample to match prior sample updates.
- Updates Twin sample to match prior sample updates.  Also refactors to match function patterns in other samples.
- Fixes [out] in iot_sample_common.h per Anton's info

### CODE REVIEW

File: sdk/samples/iot/paho_iot_hub_methods_sample.c [hunk 1]
```diff
@@ -50,9 +50,9 @@ static void parse_method_message(
     char* topic,
     int topic_len,
     MQTTClient_message const* message,
-    az_iot_hub_client_method_request* method_request);
-static void invoke_method(az_iot_hub_client_method_request const* method_request);
-static az_span ping_method(void);
+    az_iot_hub_client_method_request* out_method_request);
+static void handle_method_message(az_iot_hub_client_method_request const* method_request);
```
Comment (danewalton) [inline sdk/samples/iot/paho_iot_hub_methods_sample.c:54]:
```suggestion
static void handle_method_request(az_iot_hub_client_method_request const* method_request);
```

File: sdk/samples/iot/paho_iot_hub_methods_sample.c [hunk 2]
```diff
@@ -211,41 +213,43 @@ static void receive_method_messages(void)
...
     {
       topic_len = (int)strlen(topic);
     }
-    LOG_SUCCESS("Message #%d: Client received a message from the service.", message_count + 1);
+    IOT_SAMPLE_LOG_SUCCESS(
+        "Message #%d: Client received a Method message from the service.", message_count + 1);
 
     // Parse method message and invoke method.
     az_iot_hub_client_method_request method_request;
     parse_method_message(topic, topic_len, message, &method_request);
```
Comment (danewalton) [inline sdk/samples/iot/paho_iot_hub_methods_sample.c:242]: sorry not part of this pr but
```suggestion
    parse_method_request(topic, topic_len, message, &method_request);
```

File: sdk/samples/iot/paho_iot_hub_telemetry_sample.c [hunk 3]
```diff
@@ -168,12 +168,13 @@ static void send_telemetry_messages_to_iot_hub(void)
...
               &hub_client, NULL, telemetry_topic_buffer, sizeof(telemetry_topic_buffer), NULL)))
   {
-    LOG_ERROR("Failed to get the Telemetry publish-topic: az_result return code 0x%08x.", rc);
+    IOT_SAMPLE_LOG_ERROR("Failed to get the Telemetry topic: az_result return code 0x%08x.", rc);
     exit(rc);
   }
 
-  const char* telemetry_message_payloads[MAX_TELEMETRY_MESSAGE_COUNT] = {
-    "Message One", "Message Two", "Message Three", "Message Four", "Message Five",
+  char const* telemetry_message_payloads[MAX_TELEMETRY_MESSAGE_COUNT] = {
```
Comment (danewalton) [inline sdk/samples/iot/paho_iot_hub_telemetry_sample.c:175]: curious why you changed these

File: sdk/samples/iot/paho_iot_hub_methods_sample.c (hunk 2 above)
Comment (momuno) [reply sdk/samples/iot/paho_iot_hub_methods_sample.c:242]: it still fits.  all part of finalizing the samples.

File: sdk/samples/iot/paho_iot_hub_telemetry_sample.c (hunk 3 above)
Comment (momuno) [reply sdk/samples/iot/paho_iot_hub_telemetry_sample.c:175]: Now when I print out the payload for the message, it looks like json formatted telemetry.  I was trying to mimic the telemetry format found in twin sample.

File: sdk/samples/iot/paho_iot_hub_methods_sample.c (hunk 2 above)
Comment (momuno) [reply sdk/samples/iot/paho_iot_hub_methods_sample.c:242]: hmmm, after looking at it, I think it should stay with message.  The message is what is being parse, not the method_request.  The method_request will be initialized from info parsed from the message.
